src/Couchbase.Lite.Shared/Query/QueryExpression.cs

```diff
@@ -146,7 +146,7 @@ public IExpression Between(IExpression expression1, IExpression expression2)
 
         public IExpression Collate(ICollation collation)
         {
-            var col = Misc.TryCast<ICollation, QueryCollation>(collation);
+            var col = Misc.TryCast<ICollation, QueryCollation>(new QueryCollation((QueryCollation)collation));
```

Comment:
Leave `col` as it is.  It's casting and checking for a valid object.  Then create a copy based on `col`